docs/developer/crd-versioning.md

```diff
@@ -0,0 +1,54 @@
+# CRD Versioning
+
+This document discusses how we version our CustomResourceDefinitions (CRDs) that define Policies, Filters, or other configuration objects. NGINX Gateway Fabric distributes a few CRDs, such as NginxGateway, NginxProxy, ClientSettingsPolicy, and ObservabilityPolicy. More will inevitably be added over time.
```

Comment:
```suggestion
This document discusses how we version our CustomResourceDefinitions (CRDs) that define Policies, Filters, or other configuration objects. 

NGINX Gateway Fabric currently distributes the following CRDs: 

- NginxGateway
- NginxProxy,
- ClientSettingsPolicy
- ObservabilityPolicy.
```
If the list is going to get bigger in the future, swapping it to be bullets will help with readability.

Best to stick to statements about the current version of the project without making promises about the future.

Reply:
My hope was that we don't want to have to update this doc every time we add a new CRD, and just supply a sample list of some that we distribute.